src/components/button/button.test.js

```diff
@@ -0,0 +1,191 @@
+      (booleanState, cssValue) => {
+        CypressMountWithProviders(
+          <Button noWrap={booleanState}> Long long long long long text </Button>
+        );
+
+        if (booleanState === true) {
+          buttonDataComponent()
+            .children()
+            .children()
+            .should("have.value", cssValue, "wrap");
+        } else {
+          buttonDataComponent()
+            .children()
+            .children()
+            .should("not.have.value", cssValue, "wrap");
+        }
+      }
+    );
+
+    it("should check Button is disabled", () => {
+      CypressMountWithProviders(<ButtonDifferentTypes disabled />);
+
+      buttonDataComponent()
+        .eq(positionOfElement("first"))
+        .should("be.disabled")
+        .and("have.attr", "disabled");
+      buttonDataComponent()
+        .eq(positionOfElement("second"))
+        .should("be.disabled")
+        .and("have.attr", "disabled");
+      buttonDataComponent()
+        .eq(positionOfElement("third"))
+        .should("be.disabled")
+        .and("have.attr", "disabled");
+      buttonDataComponent()
+        .eq(positionOfElement("fourth"))
+        .should("be.disabled")
+        .and("have.attr", "disabled");
+    });
+
+    it("should check Button is enabled", () => {
+      CypressMountWithProviders(<ButtonDifferentTypes />);
+
+      buttonDataComponent().eq(positionOfElement("first")).should("be.enabled");
+      buttonDataComponent()
+        .eq(positionOfElement("second"))
+        .should("be.enabled");
+      buttonDataComponent().eq(positionOfElement("third")).should("be.enabled");
+      buttonDataComponent()
+        .eq(positionOfElement("fourth"))
+        .should("be.enabled");
+    });
+  });
+
+  describe("Check events for Button component", () => {
+    it("should call onClick event", () => {
+      const callback = cy.stub();
+
+      CypressMountWithProviders(<Button onClick={callback} />);
+
+      buttonDataComponent()
+        .click({ force: true })
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should trigger onBlur event", () => {
+      const callback = cy.stub();
+
+      CypressMountWithProviders(<Button onBlur={callback} />);
+
+      buttonDataComponent()
+        .focus()
+        .blur({ force: true })
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+
+    it("should call the onKeyDown", () => {
+      const callback = cy.stub();
+      CypressMountWithProviders(<Button onKeyDown={callback} />);
+
+      buttonDataComponent()
+        .trigger("keydown", { force: true }, keyCode("rightarrow"))
+        .then(() => {
+          // eslint-disable-next-line no-unused-expressions
+          expect(callback).to.have.been.calledOnce;
+        });
+    });
+  });
+});
```

Comment:
Why are the onChange and onFocus actions not tested here if onClick, onBlur and onKeyDown are?

Reply:
We could do a test for onFocus, but not onChange. onChange is something different with more complicated scenarios. It is hard to test onChange event in button, we discussed it and I decided to leave it for now.

Reply:
I think it should be similar to `onClick` test :)

Reply:
onChange test will fail if I apply the same tests

Reply:
After discussion we've found that onChange does not apply to Button so it's correct that you did not include it.